fix(hogql): documentation improvements

mariusandra opened the pull request:
## Changes

First batch of docs improvements.

- I changed "PostHog SQL" back to HogQL
- I changed "JSON" to "Custom" to make people want to click it less... and to better explain what to expect.
- Still kept "SQL" as the insight type, because it looks better.
- I added lines of text above the "SQL" and "Custom" editors

![image](https://user-images.githubusercontent.com/53387/226581976-eba953a6-7236-4903-87ca-3db3266bffb0.png)

![image](https://user-images.githubusercontent.com/53387/226581948-658ee7c7-d5b7-4238-9565-ff82b803c878.png)

![image](https://user-images.githubusercontent.com/53387/226582037-f79f5b7d-1a30-46b6-adfd-854c41fec23a.png)


## How did you test this code?

Visually

mariusandra committed:
fix(hogql): documentation improvements

mariusandra committed:
dot dot dot

mariusandra committed:
location location location

yakkomajuri commented on frontend/src/scenes/insights/Insight.tsx:
```diff
@@ -75,14 +82,28 @@ export function Insight({ insightId }: InsightSceneProps): JSX.Element {
+                            <div className="inline-block align-middle">
+                                <OpenEditorButton query={query} type="secondary" size="small" className="ml-1 mr-1" />
+                            </div>
+                            open any part of PostHog as a new insight. Read the{' '}
```
what is a part of posthog?

fuziontech committed:
Update frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx

Co-authored-by: Yakko Majuri <38760734+yakkomajuri@users.noreply.github.com>

fuziontech reviewed (approved):
🚢 it

mariusandra committed:
Merge branch 'master' into hogql-insight-improvements

mariusandra committed:
Merge branch 'master' into hogql-insight-improvements

mariusandra committed:
"part of posthog" -> "data table"

mariusandra commented:
Ok, we removed the "custom" option from the menu:
<img width="610" alt="image" src="https://user-images.githubusercontent.com/53387/226876778-4d612c0e-e49b-4e11-a2d9-5440d86be738.png">

 I also changed "part of posthog" to "data table", but that's something we should discuss at the standup anyway.

pauldambra committed:
Merge branch 'master' into hogql-insight-improvements

pauldambra committed:
move examples and help into query editor component

pauldambra committed:
remove debug change

pauldambra committed:
less rushed though

pauldambra commented on frontend/src/scenes/saved-insights/SavedInsights.tsx:
```diff
@@ -73,49 +73,49 @@ export interface InsightTypeMetadata {
-        name: 'JSON',
-        description: 'Build custom insights with our JSON query language',
+        name: 'Custom',
+        description: 'Save any data table as an insight, and edit the JSON directly.',
```
```suggestion
        description: 'Save components powered by our JSON query language.',
```

pauldambra committed:
Update frontend/src/scenes/saved-insights/SavedInsights.tsx

pauldambra committed:
use a lemon select in the debug view as well
